webapp/views.py

```diff
@@ -495,7 +540,37 @@ def advantage_view():
                     subscription["purchasedProductListings"]
                 )
 
+            for subscription in yearly_subscriptions:
+                for purchase in subscription["purchasedProductListings"]:
+                    listing_id = purchase["productListing"]["id"]
+                    purchased_product_id = listings_to_products[listing_id]
+                    yearly_products[purchased_product_id][
+                        "quantity"
+                    ] = purchase["value"]
+
             for contract in account["contracts"]:
+                contract["period"] = get_contract_period(contract)
```

Comment:
It is impossible to determine the period for a contract. The best you can do is determine whether it is affected by a monthly subscription, by a yearly subscription, or both. For that, look into `monthly_subscriptions[contract.products[0]]`, and if you find anything there, it means that contract is currently affected by a monthly subscription.

The same applies to yearly, of course.

If you are interested in displaying prices for both, perhaps you could append them to an `affectedBy` list with the details (product listing object).